Initial work on authentication

VictorDelCampo opened the pull request:
RISDEV-5805

VictorDelCampo committed:
Initial work on authentication

RISDEV-5805

VictorDelCampo committed:
Add explicit ports for webapp

RISDEV-5805

SebastianRossa committed:
Add mock user for functional tests

RISDEV-5805

VictorDelCampo committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into RISDEV-5805_spring_security_oauth2

# Conflicts:
#	backend/gradle/libs.versions.toml

VictorDelCampo committed:
Second iteration authentication (nginx + fallback backend route)

RISDEV-5805

SebastianRossa committed:
Use testcontainers to spin up keycloak

RISDEV-5805

VictorDelCampo committed:
Handover to malte

RISDEV-5805

malte-laukoetter committed:
Fix local keycloak setup

We use a service called localhost to be able to address the keycloak
service as "localhost:8443" in the application.yaml of our application.
Using "keycloak:8443" is not possible as spring would then redirect the
user to "keycloak:8443" to login, but that is a host that is not
available on the host system. The spring boot oauth implementation is
very restrictive on the issuer and therefore creates errors if the hosts
used by it and the user to connect to keycloak differ.

See also:
 - https://github.com/spring-projects/spring-security/issues/14633
 - https://github.com/keycloak/keycloak/issues/29783
 - https://github.com/keycloak/keycloak/issues/24252
 - https://medium.com/@kostapchuk/integrating-keycloak-with-spring-boot-in-a-dockerized-environment-813eab1f140c

RISDEV-5805

malte-laukoetter committed:
Enable keycloak healthcheck

RISDEV-5805

SebastianRossa committed:
Fix unit tests

RISDEV-5805

SebastianRossa committed:
Ignore fallback serving index.html for api routes

RISDEV-5805

malte-laukoetter committed:
Split docker-compose into two files

One for the full application and one for just the services. We need
different network configurations for both.

RISDEV-5805

malte-laukoetter committed:
Respond with 401 for unauthorized api calls

RISDEV-5805

malte-laukoetter committed:
Redirect to login page if api calls respond with 401

RISDEV-5805

malte-laukoetter committed:
Extend vite proxy with auth urls

RISDEV-5805

malte-laukoetter commented on backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/controller/LogoutController.java:
```diff
@@ -0,0 +1,34 @@
+ */
+@RestController
+@RequestMapping("/api/v1/logout")
+public class LogoutController {
```
Do we really need this? Spring automatically creates a /logout route for this purpose

andreasphil committed:
test: include auth in E2E tests

RISDEV-5805

andreasphil committed:
Merge remote-tracking branch 'origin/main' into RISDEV-5805_spring_security_oauth2

andreasphil committed:
test: separate setup for different browsers

RISDEV-5805

malte-laukoetter committed:
Bind path parameter

To fix sonar warning, removing is not possible as regexes in the
get-mapping need to be assigned to a name

RISDEV-5805

SebastianRossa committed:
Invalidate session in keycloak on logout and add logout button

- using RP-Initiated Logout see https://docs.spring.io/spring-security/reference/servlet/oauth2/login/logout.html

RISDEV-5806

hamo225 committed:
Navigate to /logout endpoint NOT make logout api call

RISDEV-5805

SebastianRossa committed:
Add integration test that user must be logged in

RISDEV-5806

SebastianRossa committed:
Get username from endpoint

RISDEV-5806

hamo225 committed:
feat: add `useGetUserName` service and update logout button to link

- Added `useGetUserName` service to fetch logged-in user's name
- Replaced the logout button with a `RouterLink`
- Updated the navbar layout to display user details and the logout link

RISDEV-5805

hamo225 committed:
feat(router): add `/logout` route to handle logout redirection

RISDEV-5805

hamo225 committed:
feat(ui): add fallback for username in case API fails

RISDEV-5805

SebastianRossa committed:
Get oidc username from endpoint

RISDEV-6102

malte-laukoetter committed:
Remove port publishing for keycloak when using localhost-network

RISDEV-0000

VictorDelCampo committed:
Merge from main

RISDEV-0000

VictorDelCampo committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into RISDEV-5805_spring_security_oauth2

VictorDelCampo committed:
Merge from main Part II

RISDEV-0000

SebastianRossa committed:
Integrationtest get oidc username from endpoint

RISDEV-6102

hamo225 committed:
refactor(security): use constructor injection in SecurityConfig

RISDEV-5805

hamo225 committed:
test: verify username display and logout behavior in e2e tests

RISDEV-5805

hamo225 committed:
refactor: move samplesDirectory to a utility file for reuse

RISDEV-5805

hamo225 committed:
fix: update testMatch to match global-setup.ts in Playwright config

RISDEV-5805

VictorDelCampo committed:
Move login/logout e2e to separate projects

RISDEV-0000
